graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
```

Comment:
Optional makes no sense unless we have parser options, which... is just dumb. Making it a reference is only worth it if the value is very large, but this is at most 2 strings, which themselves are references. Thus in the end it only copies the references, thus the total worst case cost for this is only 2 pointers.